cedar-policy-core/src/parser/err.rs

```diff
@@ -48,6 +48,71 @@ pub enum ParseError {
     #[error("poorly formed: {0}")]
     #[diagnostic(code(cedar_policy_core::parser::to_ast_err))]
     ToAST(String),
+
+    /// Error in the CST -> AST transform, internal invariant violation.
+    #[error("poorly formed: {0}")]
+    #[diagnostic(code(cedar_policy_core::parser::to_ast_err))]
+    InternalInvariantViolation(String),
+
+    /// Error in the CST -> AST transform, too many `-`s.
+    #[error("too many -'s: {0}")]
+    #[diagnostic(code(cedar_policy_core::parser::to_ast_err))]
+    TooManyDashes(String),
+
+    /// Error in the CST -> AST transform, too many `!`s.
+    #[error("too many !'s: {0}")]
+    #[diagnostic(code(cedar_policy_core::parser::to_ast_err))]
+    TooManyBangs(String),
+
+    /// Error in the CST -> AST transform, illegal entity lookup syntax.
+    #[error("unsupproted entity lookup syntax: {0}")]
+    #[diagnostic(code(cedar_policy_core::parser::to_ast_err))]
+    UnsupportedEntityLookupSyntax(String),
```

Comment:
Maybe remove `Syntax`?

Reply:
I think it helps clarify that the syntax isn't supported. Otherwise it might sound like it's unsupported in certain circumstances but not others.